### PR TITLE
Go user language support documentation using cloudstate paradox plugin

### PR DESCRIPTION
**Pull request by marcellanz**
Fixing #25 by adding documentation using the cloudstate paradox plugin and beeing separated from the main Cloudstate repository.

- Compiles and renders locally.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+    - stage: deploy docs
+      name: deploy snapshot docs to cloudstate.io
+      if: branch = master AND type = push
+      language: scala
+      script: cd docs && sbt deploy
+
+env:
+  global:
+    # encrypted with: travis encrypt --pro -r cloudstateio/python-support DEPLOY_DOCS_TOKEN=<token>
+    secure: "lALDa5xUg8uUAoxmn1XxwycRa+xp1IcohMuk+1i+GKI7BX60SI8ueHo+FZU3H+PF53CjtxA1UefsAFwz8smtWiePWbqalbLVMx7N2bt7tLjA/lhQC2+vEZu3LPMYGkxE/XwQ5Zr9KLgoci33USu3R6mxBgg3EbcVSSkOyBhGgH2VMIxktPf5Ae64khlhKJtkBlPgktuKytR6bESFMtdQ2v+wXQm0gQgTq85OhMc0hyixSO3JIshfEXd9Ywkfjj7j1njba/yb19tVeKh9GZh+PFLFp84CJwPE2R0ZoWGYMACsRwUQNmLaFb9y+GjTryanYlzbz8FGOVeL3INw7cDJriLopSYGEssiNGSZQh2U8+01N79VohzzBR+XSx/uw8Gly7C51rI7wFs0VbkUtPGFSbAcDGK6OqQ2fFsL1qwYIU+uLWpt2dtcxtbRsDpE3GL//8jUGLE6r6ov0b0W262PNbrmCaFdm9JzhjaypjvjMNBebmi4ezm8RS22lRT5wykJOzuifBTubgsSx7K2TIuTQrGv7zTOjnWoaSUy8nLiAuFp/WmCRQEaAZ9LXzLigzrnI+LtwPDnwvNP5xF+jmRnGxk27pFipDKmZZW/IXFR0uij14U7ewIFAWWtl/g2Fyi47JMXMLwR1IdBJ3QBZBlP+EX4SUvwg8YR1qUGRQbxXDs="
```
Comment (pvlugter) [inline .travis.yml]: Here's an encrypted token for this repo:

```suggestion
    # encrypted with: travis encrypt --pro -r cloudstateio/go-support DEPLOY_DOCS_TOKEN=<token>
    secure: "L1uMy3mR5U+0HFmjJwSD6i/av667T187DkzIlRwfTnEGPG6LoZ5EwyGC/IxbKbV2EyUmxt7zDO+OXvsiZx0l0mXZ5ozAKdTB7eb/ujhinwnvB82OtKu8ejvZqI0ooPO+D1RlxqZF5HCX69L0DfTjhytkqLUIlN4CTtSO4ElJ/1Ottg0c3qiNUYvNdFqsKsF7Dq7EJU8G/vZ6o1WCUJzawLshOGDQ9jSL14QnKquFr/cMXHcg3GgiPVl2NFWAld0fOuNpYqUy2X3O7V4/FOJGe+btOao4xQyVsn9V/1VFsqRCyp1uy+kPbQbA0f+fBY3CnDpLKHrhYv8TDdfP9tbUEIhGikc+aqYeGOP6+4vPK/Xi/p2pZEZErPcBfc7CoJECNG4csMjq0GGdhp6UYJO1JPMVrLFZsWLoxPiXTkreQ+Gj4+Jgia20xfcUR/tkZrCQMnkySfFPHK48g2VOadfIZNz6hbdMGrZUarMrWC/hi3DIUdoT0Wn6y5LuP8goAbIBI/+KXvECIdiqTHu7WIwtdesTIZJvHhdp44zec8Teb0Bo6uYoIrKXvfTQbgvTzMzZO8DZ1L21OL7EKD9H6rvFabZXz5I2i9695Q6gkQYmK6Twg6CK1p/aZXmJWN8QXceXPavl3iNvC2fMSFxZvHBuqAjknCQlmTH7/BKyEw5AQ/Q="
```

File: docs/src/main/paradox/gettingstarted.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+To get support for the Cloudstate event emission the Cloudstate entity should embed the `cloudstate.EventEmitter` type. The EventEmitter allows the entity to emit events during the handling of commands.
+
+Second, during registration of the entity, an entity factory function has to be provided so Cloudstate gets to know how to create and initialize an event sourced entity. 
+
+@@snip [shoppingcart.go]($base$/cloudstate/eventsourced.go) { #event-sourced-entity-func }
+
+This entity factory function returns a `cloudstate.Entity` which itself is a composite interface of a `cloudstate.CommandHandler` and a `cloudstate.EventHandler`. Every event sourced entity has to implement these two interfaces.
+
+Link to @extref:[event sourcing](cloudstate:user/features/eventsourced.html)
```
Comment (pvlugter) [inline docs/src/main/paradox/gettingstarted.md]: Do you want this link here?

File: docs/src/main/paradox/serialization.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Serialization
+
+Cloudstate functions serve gRPC interfaces, and naturally the input messages and output messages are protobuf messages that get serialized to the protobuf wire format. However, in addition to these messages, there are a number of places where Cloudstate needs to serialize other objects, for persistence and replication. This includes:
+
+* Event sourced @extref[events and snapshots](cloudstate:user/lang/go/eventsourced.html#persistence-types-and-serialization).
```
Comment (pvlugter) [inline docs/src/main/paradox/serialization.md]: Link to the old go docs, should be a local ref?

```suggestion
* Event sourced @ref[events and snapshots](eventsourced.md#persistence-types-and-serialization).
```

**Review by pvlugter — APPROVED** [review]
Nice, LGTM. Added the encrypted deploy token. Great to see the code snippets pulled from the tested TCK sample app.

File: docs/src/main/paradox/gettingstarted.md (hunk 2 above)
Comment (marcellanz) [reply docs/src/main/paradox/gettingstarted.md]: no, thanks.

**Comment by marcellanz** [comment]
@pvlugter thanks for the catches.

**Comment by marcellanz** [comment]
@pvlugter that one worked: https://cloudstate.io/docs/go/snapshot/

**Comment by pvlugter** [comment]
Cool. I've published these docs as version 0.1.0: https://cloudstate.io/docs/go/current/
Alias of: https://cloudstate.io/docs/go/0.1.0/

It will publish automatically on new tags in the future.

Updated the index page at https://cloudstate.io/docs/

@marcellanz do you want to delete the old docs from cloudstate repo now, and just have `user/lang/go/index.md` that links to https://cloudstate.io/docs/go/current/ ?

**Comment by marcellanz** [comment]
@pvlugter thanks, great! 

> do you want to delete the old docs from cloudstate repo now

sure, I'll send a PR for that.
Can we have the link in the menu be linked directly to https://cloudstate.io/docs/go/current/ with paradox instead of having `user/lang/go/index.md` and inside of that again a link?

**Comment by marcellanz** [comment]
… or we could have no TOC for `../user/lang/index.md` and having them linked in the `@@@ index`within this page once all language support docs are external?

**Comment by pvlugter** [comment]
I don't think we've got around to allowing external links in the navigation yet. Issues still open, such as: https://github.com/lightbend/paradox/issues/286

**Comment by marcellanz** [comment]
I liked the issue. Will link within `user/lang/go/index.md` then to https://cloudstate.io/docs/go/current/.

**Pull request merged**